### ISSUE TITLE
seedFromUri - AWS - use default credentials provider chain

### ISSUE THREAD

**Issue by ikwattro** [neo4j/neo4j#13038]
It seems the seedFromUri and the s3:// protocol requires the usage of access key id and access key secret.

Is it possible to use other methods from the credentials provider chain ? 

Ref : https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/auth/DefaultAWSCredentialsProviderChain.html

Reason : 

- Our EC2 instances are granted access to s3 buckets
- We use AWS SSO for individuals access to buckets

Basically, we don't use access key id/secrets for individuals or applications and leverage safer methods available in AWS.

**Comment by tonbut** [comment]
Hi @ikwattro, there are plans to extend coverage of seed providers to cover more use cases including more security models and more cloud providers. But unfortunately at the moment the build in mechanism only works as advertised. 

It possible and hopefully quite straightforward to code custom implementations and have them detected by Neo4j. The documentation on how to do this isn't currently part of the online documentation but i know it will be added soon. If this path is interesting to you as a faster solution than waiting for product changes please let me know and i can send you the required information.

**Comment by ikwattro** [comment]
Hi @tonbut , Thanks a lot for coming back.

Happy to hear it's possible to make custom implementations, I did some for LOAD CSV in the past so I wouldn't have much issues making it here.

If there are some examples, even non official yet, I'm on the Neo4j Inc. Slack ( Christophe Willemsen ) or email is christophe@graphaware.com , otherwise I can wait, no issues 😄 .

Thanks a bunch !

**Comment by ikwattro** [comment]
Since the documentation for creating your own seedProvided has been released ( thanks @tonbut ), I'm marking this ticket closed as it does satisfy my needs.
